File: JAVA/MbtTest.java
import java.util.Scanner;

public class MbtTest {

public static void allQuestions() {
Scanner input = new Scanner(System.in);

System.out.println("FREE PERSONALITY TEST\n");
System.out.println("Please select (A) or (B)\n");

        // ---------- Question 1 ----------
        System.out.println("1. You regularly make new friends.");
        System.out.println("(A) AGREE");
        System.out.println("(B) DISAGREE");
        String answerOne = input.nextLine().toUpperCase();

        if (!answerOne.equals("A") && !answerOne.equals("B")) {
            System.out.println("Invalid choice! Please enter A or B.");
            return;
}

        // ---------- Question 2 ----------
        System.out.println("\n2. You can easily connect with people you have just met.");
        System.out.println("(A) AGREE");
        System.out.println("(B) DISAGREE");
        String answerTwo = input.nextLine().toUpperCase();
        if (!answerTwo.equals("A") && !answerTwo.equals("B")) {
            System.out.println("Invalid choice! Please enter A or B.");
            return;
}

        // ---------- Question 3 ----------
        System.out.println("\n3. If your plans are interrupted, your top priority is to get back on track as soon as possible.");
        System.out.println("(A) AGREE");
        System.out.println("(B) DISAGREE");
        String answerThree = input.nextLine().toUpperCase();
        if (!answerThree.equals("A") && !answerThree.equals("B")) {
            System.out.println("Invalid choice! Please enter A or B.");
            return;
        }

        // ---------- Question 4 ----------
        System.out.println("\n4. You are still bothered by mistakes that you made a long time ago.");
        System.out.println("(A) AGREE");
        System.out.println("(B) DISAGREE");
        String answerFour = input.nextLine().toUpperCase();
        if (!answerFour.equals("A") && !answerFour.equals("B")) {
            System.out.println("Invalid choice! Please enter A or B.");
            return;
}

        // ---------- Question 5 ----------
        System.out.println("\n5. Your emotions control you more than you control them.");
        System.out.println("(A) AGREE");
        System.out.println("(B) DISAGREE");
        String answerFive = input.nextLine().toUpperCase();
        if (!answerFive.equals("A") && !answerFive.equals("B")) {
            System.out.println("Invalid choice! Please enter A or B.");
            return;
}

       

        // ---------- Results ----------
        System.out.println("\nThe options you selected are:\n");
        System.out.println("1. " + answerOne);
        System.out.println("2. " + answerTwo);
        System.out.println("3. " + answerThree);
        System.out.println("4. " + answerFour);
        System.out.println("5. " + answerFive);

        // Personality result (simple check)
        if (answerOne.equals("A") || answerTwo.equals("A")) {
            System.out.println("\nBased on your answers you are an Extrovert");
        } else {
            System.out.println("\nBased on your answers you are an Introvert");
        }

       
}

    public static void main(String[] args) {
        allQuestions();
}
}

File: PYTHON/mbt_Test.py
def all_questions():
    print("FREE PERSONALITY TEST\n")
    print("Please select (A) or (B)\n")

    # ---------- Question 1 ----------
    print("1. You regularly make new friends.")
    print("(A) AGREE")
    print("(B) DISAGREE")
    answer_one = input("Your answer: ").upper()
    if answer_one not in ["A", "B"]:
        print("Invalid choice! Please enter A or B.")
        return

    # ---------- Question 2 ----------
    print("\n2. You can easily connect with people you have just met.")
    print("(A) AGREE")
    print("(B) DISAGREE")
    answer_two = input("Your answer: ").upper()
    if answer_two not in ["A", "B"]:
        print("Invalid choice! Please enter A or B.")
        return

    # ---------- Question 3 ----------
    print("\n3. If your plans are interrupted, your top priority is to get back on track as soon as possible.")
    print("(A) AGREE")
    print("(B) DISAGREE")
    answer_three = input("Your answer: ").upper()
    if answer_three not in ["A", "B"]:
        print("Invalid choice! Please enter A or B.")
        return

    # ---------- Question 4 ----------
    print("\n4. You are still bothered by mistakes that you made a long time ago.")
    print("(A) AGREE")
    print("(B) DISAGREE")
    answer_four = input("Your answer: ").upper()
    if answer_four not in ["A", "B"]:
        print("Invalid choice! Please enter A or B.")
        return

    # ---------- Question 5 ----------
    print("\n5. Your emotions control you more than you control them.")
    print("(A) AGREE")
    print("(B) DISAGREE")
    answer_five = input("Your answer: ").upper()
    if answer_five not in ["A", "B"]:
        print("Invalid choice! Please enter A or B.")
        return

    # ---------- Display Results ----------
    print("\nThe options you selected are:\n")
    print("1.", answer_one)
    print("2.", answer_two)
    print("3.", answer_three)
    print("4.", answer_four)
    print("5.", answer_five)

    # Personality result (simple check)
    if answer_one == "A" or answer_two == "A":
        print("\nBased on your answers you are an Extrovert")
    else:
        print("\nBased on your answers you are an Introvert")


all_questions()